src/store/api/chat.ts

```diff
@@ -17,11 +17,58 @@ export const chatApi = createApi({
       query: () => `/connection`,
       providesTags: () => [{ type: CHAT_TAG.CONNECTIONS }],
     }),
+
+    createRoom: builder.mutation<void, RoomRequest>({
+      query: (body) => ({ url: "/room", method: "POST", body }),
+      invalidatesTags: [CHAT_TAG.ROOMS],
+    }),
+
+    addUsersToRoom: builder.mutation<
+      void,
+      { roomId: string; userids: string[] }
+    >({
+      query: ({ roomId, userids }) => ({
+        url: `/room/${roomId}/add-users`,
+        method: "PATCH",
+        body: { userids },
+      }),
+      invalidatesTags: [CHAT_TAG.ROOMS],
```

Comment:
no need to invalidate  tags  on  any  changes  to  rooms  cause  data would  be  updated  through socket  connection